npm/qsharp/src/shared/circuit.ts

```diff
@@ -3,23 +3,130 @@
 
 import { Register } from "./register.js";
 
+/**
+ * Current format version.
+ */
+export const CURRENT_VERSION = 1;
+
+export interface CircuitGroup {
+  circuits: Circuit[];
+  version?: number;
+}
+
 /**
  * Circuit to be visualized.
  */
 export interface Circuit {
   /** Array of qubit resources. */
   qubits: Qubit[];
-  operations: Operation[];
+  componentGrid: ComponentGrid;
 }
 
+export type ComponentGrid = Column[];
+
+export interface Column {
+  components: Component[];
+}
+
+export type Component = Operation;
```

Comment:
Consider adding a comment describing the distinction between these two types

Reply:
done